### PR TITLE
fix: adjust context evaluation to return correct types

### PR DESCRIPTION
**Pull request by uziasferreirazup**
### Related Issues

closes issue #958 

### Description and Example

Adjust to set context always normalize value.
Adjust SendRequest to just normalize value and create an implicit context.
Remove method `trytodeserialize` and adjust the evaluation of the expression to return correct type 

### Checklist

Please, check if these important points are met using `[x]`:

- [X] I read the [PR Guide] and followed the process outlined there for submitting this PR.
- [X] I avoided _breaking changes_ by not changing public APIs that people rely on. 
- [X] I am willing to follow-up on review comments in a timely manner.

[PR Guide]: https://github.com/ZupIT/beagle/blob/master/doc/contributing/pull_requests.md

### CODE REVIEW

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/ZupIT/beagle/blob/c9ab35418e22355d1ca7a276ed025a5c41ecb792/android/beagle/src/main/java/br/com/zup/beagle/android/context/ContextDataEvaluation.kt#L53">android/beagle/src/main/java/br/com/zup/beagle/android/context/ContextDataEvaluation.kt#L53</a> - Function evaluateBindExpressions has 3 return statements which exceeds the limit of 2.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_android">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: android/beagle/src/main/java/br/com/zup/beagle/android/components/form/core/FormSubmitter.kt [hunk 1]
```diff
@@ -86,14 +86,13 @@ internal class FormSubmitter(
...
 
-    private fun createUrl(form: FormRemoteAction, formsValue: Map<String, String>)
-        = if (form.method == FormMethodType.GET || form.method == FormMethodType.DELETE)
-            formsValue.filterValues {
-                isFormsValueValid(it)
-            }.toList().fold(Uri.parse(form.path).buildUpon()) { path, param ->
-                path.appendQueryParameter(param.first, param.second)
-            }.build().toString()
-        else form.path
+    private fun createUrl(form: FormRemoteAction, formsValue: Map<String, String>) = if (form.method == FormMethodType.GET || form.method == FormMethodType.DELETE)
```
Comment (botbeaglezup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/components/form/core/FormSubmitter.kt:89]: <table data-meta="generated_by_android">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/4d8aa400507297e6a523decddba9a85e1ac79cc1/android/beagle/src/main/java/br/com/zup/beagle/android/components/form/core/FormSubmitter.kt#L89"></span>Line detected that is longer than the defined maximum line length in the code style.</td>
    </tr>
  </tbody>
</table>

**Review by matheusribeirozup — APPROVED** [review]

File: android/beagle/src/test/java/br/com/zup/beagle/android/context/ContextDataEvaluationTest.kt [hunk 2]
```diff
@@ -383,10 +382,10 @@ internal class ContextDataEvaluationTest : BaseTest() {
     @Test
     fun evaluateContextBindings_with_operation_should_evaluate_contains_operation() {
```
Comment (viniciusguardieirozup) [inline android/beagle/src/test/java/br/com/zup/beagle/android/context/ContextDataEvaluationTest.kt:383]: Please, use the test name defined `GIVEN an operation WHEN evaluateContextBindings called SHOULD evaluate contains operation`

File: android/beagle/src/test/java/br/com/zup/beagle/android/context/ContextDataEvaluationTest.kt [hunk 3]
```diff
@@ -399,18 +398,19 @@ internal class ContextDataEvaluationTest : BaseTest() {
     @Test
     fun evaluateContextBindings_with_operation_should_throw_error_insert_operation_index_out_of_bound() {
```
Comment (viniciusguardieirozup) [inline android/beagle/src/test/java/br/com/zup/beagle/android/context/ContextDataEvaluationTest.kt:399]: same here

File: android/beagle/src/test/java/br/com/zup/beagle/android/utils/ActionExtensionsKtTest.kt [hunk 4]
```diff
@@ -262,6 +261,40 @@ class ActionExtensionsKtTest : BaseTest() {
...
+
+        // When
+        val actualValue = action.evaluateExpression(rootView, bindView, value)
+
+        // Then
+        assertEquals(contextValue, actualValue as Int)
+    }
+
     @Test
     fun evaluateExpression_should_evaluate_expression_of_type_boolean() {
```
Comment (viniciusguardieirozup) [inline android/beagle/src/test/java/br/com/zup/beagle/android/utils/ActionExtensionsKtTest.kt:299]: same here

File: android/beagle/src/test/java/br/com/zup/beagle/android/utils/ActionExtensionsKtTest.kt [hunk 5]
```diff
@@ -449,6 +482,34 @@ class ActionExtensionsKtTest : BaseTest() {
...
+        // When
+        val actualValue = action.evaluateExpression(rootView, contextView2, value)
+
+        // Then
+        assertTrue(actualValue is JSONObject)
+        assertEquals("""{"value2":${context2.value},"value1":${context1.value}}""", actualValue.toString())
+    }
+
     @Test
     fun evaluateExpression_should_return_evaluated_value_from_implicit_context_and_normal_context() {
```
Comment (viniciusguardieirozup) [inline android/beagle/src/test/java/br/com/zup/beagle/android/utils/ActionExtensionsKtTest.kt:514]: same here

File: android/beagle/src/test/java/br/com/zup/beagle/android/utils/ActionExtensionsKtTest.kt [hunk 6]
```diff
@@ -513,6 +574,23 @@ class ActionExtensionsKtTest : BaseTest() {
...
+
+        // When
+        val actualValue = secondAction.evaluateExpression(rootView, bindView, bind)
+
+        // Then
+        assertEquals(implicitContextValue, actualValue)
+    }
+
     @Test
     fun evaluateExpression_should_return_cached_result_if_there_is() {
```
Comment (viniciusguardieirozup) [inline android/beagle/src/test/java/br/com/zup/beagle/android/utils/ActionExtensionsKtTest.kt:595]: same here

**Review by viniciusguardieirozup — CHANGES_REQUESTED** [review]

**Review by viniciusguardieirozup — APPROVED** [review]

File: android/beagle/src/main/java/br/com/zup/beagle/android/context/ContextDataEvaluation.kt [hunk 7]
```diff
@@ -56,35 +56,76 @@ internal class ContextDataEvaluation(
...
-                        expressionToken,
-                        bind,
-                        evaluatedExpressions,
-                        evaluatedBindings
-                    )
+    private fun deserializeExpression(
+        bind: Bind.Expression<*>,
+        expressions: List<ExpressionToken>,
+        type: Type?,
+        response: Any?): Any? {
```
Comment (tuliopereirazup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/context/ContextDataEvaluation.kt:68]: break this parentheses to follow Kotlin

**Pull request merged**